packages/core/src/mcp/oauth-provider.ts

```diff
@@ -576,18 +579,27 @@ export class MCPOAuthProvider {
    * @param serverName The name of the MCP server
    * @param config OAuth configuration
    * @param mcpServerUrl Optional MCP server URL for OAuth discovery
+   * @param messageHandler Optional handler for displaying user-facing messages
    * @returns The obtained OAuth token
    */
   static async authenticate(
     serverName: string,
     config: MCPOAuthConfig,
     mcpServerUrl?: string,
+    events?: EventEmitter,
```

Comment:
nit: I would suggest making this a required parameter so that we don't accidentally omit passing the event emitter and regress the way this data is logged.